src/parse.par

```diff
@@ -712,7 +712,7 @@ production<PROD*>	:	line_number rhs_opt:rhs code_opt_dup:act
 
 rhs_opt<PROD*>		:	rhs
 
-					|	[*	@@ = create_production( parser,
+					|	[*	@@ = current_prod = create_production( parser,
```

Comment:
This is the real fix for the bug in #22, and I think this is all required. Thanks! :+1: